From a2ce7d7ce536e66f5932a11b3e25c9817bd4b48c Mon Sep 17 00:00:00 2001
From: Hamad Al Marri <hamad.s.almarri@gmail.com>
Date: Sat, 13 Nov 2021 06:44:47 +0300
Subject: [PATCH 04/20] Make child runs first in fork by default. It can be
 switched back to parent run first via sysctl `sysctl
 kernel.sched_child_runs_first=0`

---
 kernel/sched/bs.c | 6 +++++-
 kernel/sched/bs.h | 2 +-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/kernel/sched/bs.c b/kernel/sched/bs.c
index 38d385e45..778a538b3 100644
--- a/kernel/sched/bs.c
+++ b/kernel/sched/bs.c
@@ -1003,9 +1003,13 @@ static void task_fork_fair(struct task_struct *p)
 	cfs_rq = task_cfs_rq(current);
 
 	curr = cfs_rq->curr;
-	if (curr)
+	if (curr) {
 		update_curr(cfs_rq);
 
+		if (sysctl_sched_child_runs_first)
+			resched_curr(rq);
+	}
+
 	rq_unlock(rq, &rf);
 }
 
diff --git a/kernel/sched/bs.h b/kernel/sched/bs.h
index 7e6397024..28f2f4507 100644
--- a/kernel/sched/bs.h
+++ b/kernel/sched/bs.h
@@ -3,7 +3,7 @@
  * After fork, child runs first. If set to 0 (default) then
  * parent will (try to) run first.
  */
-unsigned int sysctl_sched_child_runs_first __read_mostly;
+unsigned int sysctl_sched_child_runs_first __read_mostly = 1;
 
 const_debug unsigned int sysctl_sched_migration_cost	= 500000UL;
 
-- 
2.34.1.75.gabe6bb3905

